hedera-node/src/main/java/com/hedera/services/ledger/interceptors/StakeAwareAccountsCommitsInterceptor.java

```diff
@@ -0,0 +1,335 @@
+		rewardCalculator.updateRewardChanges(account, changes);
+		final var reward = rewardCalculator.getAccountReward();
+		sideEffectsTracker.trackRewardPayment(account.number(), reward);
+		hasBeenRewarded[accountI] = true;
+	}
+
+	/**
+	 * Checks if the account is eligible for rewards.
+	 *
+	 * @param account
+	 * 		account that is being checked
+	 * @param changes
+	 * 		account property changes
+	 * @return true if rewardable, false otherwise
+	 */
+	boolean isRewardable(
+			@Nullable final MerkleAccount account,
+			@NotNull final Map<AccountProperty, Object> changes
+	) {
+		Boolean changedDecline = (Boolean) changes.get(DECLINE_REWARD);
+		return account != null
+				&& account.getStakedId() < 0
+				&& networkCtx.get().areRewardsActivated()
+				&& hasStakeFieldChanges(changes)
+				&& stakePeriodManager.isRewardable(account.getStakePeriodStart())
+				&& !Boolean.TRUE.equals(changedDecline)
+				&& (!account.isDeclinedReward() || Boolean.FALSE.equals(changedDecline));
+	}
+
+	/**
+	 * Checks and activates staking rewards if the staking funding account balance reaches threshold
+	 *
+	 * @param pendingChanges
+	 * 		account property changes
+	 */
+	void checkStakingRewardsActivation(
+			final EntityChangeSet<AccountID, MerkleAccount, AccountProperty> pendingChanges) {
+		final var newRewardBalance = rewardBalanceIncreased ? calculateNewRewardBalance(pendingChanges) : 0;
+		if (!shouldActivateStakingRewards(newRewardBalance)) {
+			return;
+		}
+		long todayNumber = stakePeriodManager.currentStakePeriod();
+
+		networkCtx.get().setStakingRewards(true);
+		stakeInfoManager.clearRewardsHistory();
+		stakeChangeManager.setStakePeriodStart(todayNumber);
+		log.info("Staking rewards is activated and rewardSumHistory is cleared");
+	}
+
+	/**
+	 * Calculates new reward balance for the account 0.0.800 and checks if staking rewards should be activated
+	 *
+	 * @param pendingChanges
+	 * 		account property changes
+	 * @return new balance of 0.0.800
+	 */
+	long calculateNewRewardBalance(
+			final EntityChangeSet<AccountID, MerkleAccount, AccountProperty> pendingChanges) {
+		long stakingFundBalance = 0;
+		long existingBalance = 0;
+		for (int i = 0; i < pendingChanges.size(); i++) {
+			if (pendingChanges.id(i).getAccountNum() == STAKING_FUNDING_ACCOUNT_NUMBER) {
+				existingBalance = pendingChanges.entity(i).getBalance();
+				final var changes = pendingChanges.changes(i);
+				stakingFundBalance += (long) changes.get(BALANCE);
+			}
+		}
+		return stakingFundBalance + existingBalance;
+	}
+
+	/**
+	 * If the balance on 0.0.800 changed in the current transaction and the balance reached above the specified
+	 * threshold activates staking rewards
+	 *
+	 * @param newRewardBalance
+	 * @return true if rewards should be activated, false otherwise
+	 */
+	boolean shouldActivateStakingRewards(final long newRewardBalance) {
+		return !rewardsActivated && rewardBalanceIncreased && (newRewardBalance >= dynamicProperties.getStakingStartThreshold());
```

Comment:
Given above suggestion to gate `checkStakingRewardsActivation()` with an `if`, can be just,
```suggestion
		return newRewardBalance >= dynamicProperties.getStakingStartThreshold();
```